src/BlockTypes/ProductGalleryPager.php

```diff
@@ -37,96 +39,128 @@ protected function get_block_type_uses_context() {
+				break;
+			}
+
+			$is_first_pager_item = 0 === $key;
+			$pager_item          = sprintf(
+				'<li class="wp-block-woocommerce-product-gallery__pager-item %2$s">%1$s</li>',
+				'dots' === $pager_display_mode ? $this->get_dot_icon( $is_first_pager_item ) : $key + 1,
+				$is_first_pager_item ? 'is-active' : ''
+			);
+			$p                   = new \WP_HTML_Tag_Processor( $pager_item );
+
+			if ( $p->next_tag() ) {
+				$p->set_attribute(
+					'data-wc-context',
+					wp_json_encode(
+						array(
+							'isSelected'  => $is_first_pager_item,
+							'woocommerce' => array( 'imageId' => strval( $product_gallery_image_id ) ),
+						)
+					)
+				);
+				$p->set_attribute(
+					'data-wc-on--click',
+					'actions.woocommerce.handleClick'
+				);
+				$p->set_attribute(
+					'data-wc-class--is-active',
+					'selectors.woocommerce.isSelected'
+				);
+				$html .= $p->get_updated_html();
+			}
+		}
+
 		return sprintf(
 			'<ul class="wp-block-woocommerce-product-gallery-pager__pager">
-				<li class="wp-block-woocommerce-product-gallery__pager-item is-active">%1$s</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">%2$s</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">%2$s</li>
+				%1$s
 			</ul>',
-			$this->get_selected_dot_icon(),
-			$this->get_dot_icon()
+			$html
 		);
 	}
 
 	/**
-	 * Returns the dot icon SVG code.
+	 * Generates an SVG dot icon with the specified opacity.
 	 *
-	 * @return string The dot icon SVG code.
+	 * @param bool $is_active Whether the dot icon should be in active state. Defaults to false.
+	 * @return string The SVG dot icon HTML.
 	 */
-	private function get_dot_icon() {
+	private function get_dot_icon( $is_active = false ) {
+		$initial_opacity = $is_active ? '1' : '0.2';
 		return sprintf(
 			'<svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
-				<circle cx="6" cy="6" r="6" fill="black" fill-opacity="0.2"/>
-			</svg>'
+				<circle cx="6" cy="6" r="6" fill="black" fill-opacity="%1$s" data-wc-bind--fill-opacity="selectors.woocommerce.pagerDotFillOpacity"  />
+			</svg>',
+			$initial_opacity
 		);
 	}
 
 	/**
-	 * Returns the selected dot icon SVG code.
+	 * Get the Interactivity API's view script handle for this block type.
 	 *
-	 * @return string The selected dot icon SVG code.
+	 * @param string $key Data to get, or default to everything.
 	 */
-	private function get_selected_dot_icon() {
-		return sprintf(
-			'<svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
-				<circle cx="6" cy="6" r="6" fill="black"/>
-			</svg>'
-		);
+	protected function get_block_type_script( $key = null ) {
+		$script = [
+			'handle'       => 'wc-' . $this->block_name . '-frontend',
+			'path'         => $this->asset_api->get_block_asset_build_path( $this->block_name . '-frontend' ),
+			'dependencies' => [ 'wc-interactivity' ],
+		];
+
+		return $key ? $script[ $key ] : $script;
```

Comment:
Is it necessary? It seems that we don't have any frontend files specific to the Product Gallery Pager.

Reply:
I merged #10736 with this PR so this part was removed because it is unnecessary